src/main/proto/ga4gh/read_service.proto

```diff
@@ -91,6 +99,52 @@ message GetReadGroupSetRequest {
   string read_group_set_id = 1;
 }
 
+// ******************  /readgroups  *********************/
+
+// This request maps to the body of `POST /readgroups/search` as JSON.
+
+
+message SearchReadGroupsRequest {
+  // The dataset to search.
+  string dataset_id = 1;
+  
+  // Only return readgroups within the provided read group set ID.
+  string read_group_set_id = 2;
+
+  // Only return read groups with this name (case-sensitive, exact match).
+  string name = 3;
+
+  //  Specifying the id of a BioSample record will return only readgroups
+  //  with the given bioSampleId.
+  string bio_sample_id = 4;
```

Comment:
SampleId has always been in ReadGroup.  The rationale for this
was not written down. Perhaps someone who was there can example.
@delagoya, @lh3, @richarddurbin ???

CH Albach <notifications@github.com> writes:
> Is there background on why bio_sample_id is per-ReadGroup and not
> per-ReadGroupSet? I don't see any discussion of this on the original PR. I
> would have thought the latter would be more natural (and would side-step this
> PR).